File: typescript/deployment-plan-app/test/deployment-plan-app.test.ts
import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import * as DeploymentPlanApp from '../lib/deployment-plan-app-stack';

test('Deployment plan created', () => {
  const app = new cdk.App();

  const stack = new DeploymentPlanApp.DeploymentPlanAppStack(app, 'MyDeploymentPlanStack');

  const template = Template.fromStack(stack);

  template.hasResourceProperties('Attini::Deploy::DeploymentPlan', {DeploymentPlan: {}});
});

File: java/deployment-plan-app/src/test/java/com/myorg/DeploymentPlanAppTest.java
 package com.myorg;

 import java.util.Map;

 import org.junit.jupiter.api.Test;

 import software.amazon.awscdk.App;
 import software.amazon.awscdk.assertions.Template;
 public class DeploymentPlanAppTest {

     @Test
     public void testStack() {
         App app = new App();
         DeploymentPlanAppStack stack = new DeploymentPlanAppStack(app, "test");

         Template template = Template.fromStack(stack);

         template.hasResourceProperties("Attini::Deploy::DeploymentPlan", Map.of("DeploymentPlan", Map.of()));
     }
 }
